addons/mail/models/res_users.py

```diff
@@ -16,46 +16,23 @@ class Users(models.Model):
     _name = 'res.users'
     _inherit = ['res.users']
 
-    notification_type = fields.Selection([
-        ('email', 'Handle by Emails'),
-        ('inbox', 'Handle in Odoo')],
-        'Notification', required=True, default='email',
-        compute='_compute_notification_type', inverse='_inverse_notification_type', store=True,
-        help="Policy on how to handle Chatter notifications:\n"
-             "- Handle by Emails: notifications are sent to your email address\n"
-             "- Handle in Odoo: notifications appear in your Odoo Inbox")
-
-    _sql_constraints = [(
-        "notification_type",
-        "CHECK (notification_type = 'email' OR NOT share)",
-        "Only internal user can receive notifications in Odoo",
-    )]
-
-    @api.depends('share', 'groups_id')
-    def _compute_notification_type(self):
-        # Because of the `groups_id` in the `api.depends`,
-        # this code will be called for any change of group on a user,
-        # even unrelated to the group_mail_notification_type_inbox or share flag.
-        # e.g. if you add HR > Manager to a user, this method will be called.
-        # It should therefore be written to be as performant as possible, and make the less change/write as possible
-        # when it's not `mail.group_mail_notification_type_inbox` or `share` that are being changed.
-        inbox_group_id = self.env['ir.model.data']._xmlid_to_res_id('mail.group_mail_notification_type_inbox')
-
-        self.filtered_domain([
-            ('groups_id', 'in', inbox_group_id), ('notification_type', '!=', 'inbox')
-        ]).notification_type = 'inbox'
-        self.filtered_domain([
-            ('groups_id', 'not in', inbox_group_id), ('notification_type', '=', 'inbox')
-        ]).notification_type = 'email'
-
-        # Special case: internal users with inbox notifications converted to portal must be converted to email users
-        self.filtered_domain([('share', '=', True), ('notification_type', '=', 'inbox')]).notification_type = 'email'
+    notification_type = fields.Selection(
+        default='email',
+        inverse="_inverse_notification_type",
+        readonly=False,
+        related_sudo=False,
+        related="res_users_settings_id.notification_preference",
+        store=True,
```

Comment:
Makes no sense to have both a field on res_users and on res_users_settings; this is duplicated information. Either you do a computed on the settings tables, either you keep the current model, but duplicating information is always a bad idea.

Reply:
We want to move the `notification_type` into the `res.users.settings` but the current behavior is that user can change the `notification_type` from the preference page, which is a `res.users` form. So, there should always be a field or related field in the `res.users`.

Reply:
But why ? You change is not motivated except a bug that seems unrelated to the current model. You may also keep current code and simply make a related on user_id.notification_type in settings model, seems easier.

Reply:
Indeed, at first I think it is more logical to put most of the logic in the `res.users.settings` side because it seems more logical (personal related setting). See https://github.com/odoo/odoo/pull/145041#discussion_r1416977573 I would try to do the compute in the `res.users.settings` side and keep the logic in the `res.users` now and do the sync in the setting and see if it is easier to do in that way

Reply:
It is probably less change :) I suppose it is a one-to-one relationship between a user and a settings ? Should not be that hard, and that way we keep most of current code and avoid regressions.